website/src/pages/api/subscriptions/[...path].ts

```diff
@@ -0,0 +1,74 @@
+import { getSession } from 'auth-astro/server';
+import { getDashboardsConfig } from '../../../config.ts';
+import type { Session } from '@auth/core/types';
+
+const API_PATHNAME_LENGTH = '/api'.length;
+
+export async function ALL({ request }: { request: Request }) {
+    const session = await getSession(request);
+
+    if (!session || !session.user?.id) {
+        return getUnauthorizedResponse(request.url);
+    }
+
+    const backendUrl = getBackendUrl(request, session);
+
+    try {
+        const response = await fetch(backendUrl, {
+            method: request.method,
+        });
+
+        return new Response(response.body, {
+            status: response.status,
+            headers: response.headers,
+        });
+    } catch (error) {
+        return getInternalErrorResponse(request.url);
+    }
+}
+
+function getBackendUrl(request: Request, session: Session) {
+    const backendEndpoint = new URL(request.url).pathname.slice(API_PATHNAME_LENGTH);
+    const backendUrl = new URL(backendEndpoint, getDashboardsConfig().backend.url);
+    backendUrl.searchParams.set('userId', session.user?.id || '');
+    return backendUrl;
+}
+
+type BackendError = {
+    title: string;
+    detail: string;
+    status: number;
+    instance: string;
+};
+
+const getUnauthorizedResponse = (requestUrl: string) => {
+    const response: BackendError = {
+        title: 'Unauthorized',
+        detail: "You're not authorized to access this resource",
+        status: 401,
+        instance: requestUrl,
+    };
+
+    return new Response(JSON.stringify({ response }), {
+        status: 401,
+        headers: {
+            'Content-Type': 'application/json',
+        },
+    });
+};
+
+const getInternalErrorResponse = (requestUrl: string) => {
+    const response: BackendError = {
+        title: 'Internal Server Error',
+        detail: "Couldn't fetch data from the backend",
```

Comment:
Does the `error` contain more detail about why it fails? It would be good to at least log it somehow, or even write it to the response.

Reply:
Currently, this is only the fetch failed error. The other errors will be forwarded from the backend directly, so does it make sense to add the detail here?

Reply:
Maybe a `console.error(error)` on the calling side of this function already does the job.
(We should think of a logging solution for the website...)